src/test/java/org/opensearch/knn/index/query/KNNWeightTests.java

```diff
@@ -0,0 +1,388 @@
+    private static final Set<String> SEGMENT_FILES_FAISS = Set.of("_0.cfe", "_0_2011_target_field.faissc");
+    private static final String CIRCUIT_BREAKER_LIMIT_100KB = "100Kb";
+
+    private static final Map<Integer, Float> DOC_ID_TO_SCORES = Map.of(10, 0.4f, 101, 0.05f, 100, 0.8f, 50, 0.52f);
+
+    private static MockedStatic<NativeMemoryCacheManager> nativeMemoryCacheManagerMockedStatic;
+    private static MockedStatic<JNIService> jniServiceMockedStatic;
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        final KNNSettings knnSettings = mock(KNNSettings.class);
+        final MockedStatic<KNNSettings> knnSettingsMockedStatic = mockStatic(KNNSettings.class);
+        when(knnSettings.getSettingValue(eq(KNNSettings.KNN_MEMORY_CIRCUIT_BREAKER_ENABLED))).thenReturn(true);
+        when(knnSettings.getSettingValue(eq(KNNSettings.KNN_MEMORY_CIRCUIT_BREAKER_LIMIT))).thenReturn(CIRCUIT_BREAKER_LIMIT_100KB);
+        when(knnSettings.getSettingValue(eq(KNNSettings.KNN_CACHE_ITEM_EXPIRY_ENABLED))).thenReturn(false);
+        when(knnSettings.getSettingValue(eq(KNNSettings.KNN_CACHE_ITEM_EXPIRY_TIME_MINUTES))).thenReturn(TimeValue.timeValueMinutes(10));
+
+        final ByteSizeValue v = ByteSizeValue.parseBytesSizeValue(
+            CIRCUIT_BREAKER_LIMIT_100KB,
+            KNNSettings.KNN_MEMORY_CIRCUIT_BREAKER_LIMIT
+        );
+        knnSettingsMockedStatic.when(KNNSettings::getCircuitBreakerLimit).thenReturn(v);
+        knnSettingsMockedStatic.when(KNNSettings::state).thenReturn(knnSettings);
+        knnSettingsMockedStatic.when(KNNSettings::isKNNPluginEnabled).thenReturn(true);
+
+        jniServiceMockedStatic = mockStatic(JNIService.class);
+        nativeMemoryCacheManagerMockedStatic = mockStatic(NativeMemoryCacheManager.class);
+
+        final NativeMemoryCacheManager nativeMemoryCacheManager = mock(NativeMemoryCacheManager.class);
+        final NativeMemoryAllocation nativeMemoryAllocation = mock(NativeMemoryAllocation.class);
+        when(nativeMemoryCacheManager.get(any(), anyBoolean())).thenReturn(nativeMemoryAllocation);
+
+        nativeMemoryCacheManagerMockedStatic.when(NativeMemoryCacheManager::getInstance).thenReturn(nativeMemoryCacheManager);
+
+        final MockedStatic<PathUtils> pathUtilsMockedStatic = mockStatic(PathUtils.class);
+        final Path indexPath = mock(Path.class);
+        when(indexPath.toString()).thenReturn("/mydrive/myfolder");
+        pathUtilsMockedStatic.when(() -> PathUtils.get(anyString(), anyString())).thenReturn(indexPath);
+    }
+
+    @SneakyThrows
+    public void testQueryResultScoreNmslib() {
+        List.of(SpaceType.L2, SpaceType.L1, SpaceType.COSINESIMIL, SpaceType.INNER_PRODUCT, SpaceType.LINF).forEach(space -> {
+            try {
+                testQueryScore(space::scoreTranslation, SEGMENT_FILES_NMSLIB, Map.of(SPACE_TYPE, space.getValue()));
+            } catch (Exception e) {
+                throw new RuntimeException(e);
```

Comment:
Why catch this?

Reply:
that's the syntax required in case we're using forEach. Let me change it to something less verbose and without required catch.